inc/HttpClient/HttpClient.php

```diff
@@ -143,9 +143,15 @@ public static function retry_decider( int $retries, RequestInterface $request, ?
 	 * @return int Number of milliseconds to delay.
 	 */
 	public static function retry_delay( int $retries, ?ResponseInterface $response ): int {
-		// Be default, implement a linear backoff strategy.
-		$retry_after = $retries;
-
+		// Implement an exponential backoff strategy with jitter, with the delay capped at 10s and a minimum of 1s.
+		// Full Jitter taken from https://aws.amazon.com/blogs/architecture/exponential-backoff-and-jitter/.
+		// Given we only retry 3 times, this will really never exceed 8s.
+		$retry_after_without_jitter = min( 10, 1 * pow( 2, $retries ) );
```

Comment:
Considering we only allow 3 retry attempts, and that's not overridable it means that the max time possible is 8s. I kept it at 10s just in case.

Below this, the time is picked randomly between 1 and this value (inclusive).

Reply:
So instead of

1s 2s 3s

it would be 

2s 4s 8s

as the delays. We could experiment with changing the base to 2, so it would be 4s, 8s, 16s but I think what we have now is sufficient.